semver.js

```diff
@@ -704,6 +704,42 @@ Comparator.prototype.test = function(version) {
   return cmp(version, this.operator, this.semver, this.loose);
 };
 
+Comparator.prototype.intersects = function(compOrStr, loose, platform) {
+  var rangeTmp;
+  var comp = new Comparator(compOrStr, loose);
```

Comment:
I don't think typescript has anything to do with it; it's nicer to have monomorphic APIs with or without types. By "decisions" I mean, the comparator and the range having the same `loose` setting, for example - forcing the user to make that choice means they can never be ignorant of the default.